Sources/Sentry/SentryThreadInspector.m

```diff
@@ -12,6 +18,36 @@
 
 @end
 
+typedef struct {
+    SentryCrashThread thread;
+    SentryCrashStackEntry stackEntries[MAX_STACKTRACE_LENGTH];
+    int stackLength;
+} ThreadInfo;
+
+// We need a C function to retrieve information from the stack trace in order to avoid
+// calling into async code while there is suspended threads.
+unsigned int
+getStackEntriesFromThread(SentryCrashThread thread, struct SentryCrashMachineContext *context,
+    SentryCrashStackEntry *buffer, unsigned int amount)
+{
+    sentrycrashmc_getContextForThread(thread, context, false);
+    SentryCrashStackCursor stackCursor;
+    sentrycrashsc_initWithMachineContext(&stackCursor, MAX_STACKTRACE_LENGTH, context);
+
+    unsigned int result = 0;
+    while (stackCursor.advanceCursor(&stackCursor)) {
+        if (result == amount)
+            break;
```

Comment:
`m`: Shouldn't we somehow handle this edge case? I'm not sure how likely it is that stacktraces will be cut off, but I already see complaints that the stacktrace is incomplete.

Reply:
Hmm, I'm not sure. @Swatinem, can you maybe help us here? We allocate a struct with a maximum of 100 stack entries before suspending the threads. Here we are walking the stack cursor after suspending the threads. Do you have any recommendations on what to do when the stack has more than 100 entries? Now we ignore this, and the stacktrace would miss a few entries. We can't allocate memory either.

Reply:
@armcknight, do you maybe have any ideas on how to solve the issue above ⬆️ ?

Reply:
@brustolin, what about merging this PR and creating an issue for this edge case? This is the only issue left, I believe.

Reply:
Yes. Sounds good to me.